pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -338,8 +332,8 @@ func (e *etcd) Deploy(ctx context.Context) error {
 			return err
 		}
 
-		// create peer network policy only if the shoot has a HA control plane
-		if e.hasHAControlPlane() {
+		// create peer network policy only if there are 3 replicas
+		if pointer.Int32Deref(e.values.Replicas, 0) == 3 {
```

Comment:
```suggestion
		if pointer.Int32Deref(e.values.Replicas, 0) > 1 
```

Nit: It's more generic as we always need this network policy (peer certs similarly) when more than one etcd member is involved. Please adjust other related conditions as well.